pages/_app.jsx

```diff
@@ -1,18 +1,26 @@
 import React from 'react';
-
+import { useRouter } from 'next/router';
 import { BaseLayout } from '@components/base_layout';
 import 'bootstrap/dist/css/bootstrap.min.css';
 import '@styles/globals.scss';
+import Home from '.';
 
 function App({ Component, pageProps }) {
+    const router = useRouter();
+    const isHomePage = router.pathname === '/';
```

Comment:
This is being repeated in two places so it is not DRY.

To make it DRY, we need to create a custom hook (it is actually a function this time but hooks are a special trait of React)

- Create a file named `utils/hooks.js`
- Put the following code inside it:

```js
import { useRouter } from 'next/router';

function useIsHomePage() {
    const router = useRouter();
    return router.pathname === '/';
}
```

Then you can use this function/hook here:
```suggestion
    const isHomePage = useIsHomePage();
```

Don't forget to add `import { useIsHomePage() } from '@utils/hooks';`.
Also, Remove the unused `useRouter` import.